lms/tasks/email_digests.py

```diff
@@ -46,44 +44,37 @@ def send_instructor_email_digest_tasks(self, *, batch_size, h_userids=None):
 
     with app.request_context() as request:  # pylint:disable=no-member
         with request.tm:
-            if not h_userids:
-                try:
-                    h_userids = request.db.scalars(
-                        select(User.h_userid)
-                        .select_from(User)
-                        .distinct()
-                        .join(ApplicationInstance)
-                        .join(AssignmentMembership)
-                        .join(LTIRole)
-                        .outerjoin(
-                            EmailUnsubscribe,
-                            and_(
-                                User.h_userid == EmailUnsubscribe.h_userid,
-                                EmailUnsubscribe.tag
-                                == EmailUnsubscribe.Tag.INSTRUCTOR_DIGEST,
-                            ),
-                        )
-                        .where(
-                            ApplicationInstance.settings["hypothesis"][
-                                "instructor_email_digests_enabled"
-                            ].astext
-                            == "true",
-                            LTIRole.type == "instructor",
-                            # No EmailUnsubscribes
-                            EmailUnsubscribe.id.is_(None),
-                        )
-                    ).all()
-                except Exception as exc:  # pylint:disable=broad-exception-caught
-                    LOG.exception(exc)
-                    report_exception(exc)
-                    self.retry(countdown=_retry_countdown(self.request))
+            h_userids = request.db.scalars(
+                select(User.h_userid)
+                .select_from(User)
+                .distinct()
+                .join(ApplicationInstance)
+                .join(AssignmentMembership)
+                .join(LTIRole)
+                .outerjoin(
+                    EmailUnsubscribe,
+                    and_(
+                        User.h_userid == EmailUnsubscribe.h_userid,
+                        EmailUnsubscribe.tag == EmailUnsubscribe.Tag.INSTRUCTOR_DIGEST,
+                    ),
+                )
+                .where(
+                    ApplicationInstance.settings["hypothesis"][
+                        "instructor_email_digests_enabled"
+                    ].astext
+                    == "true",
+                    LTIRole.type == "instructor",
+                    # No EmailUnsubscribes
+                    EmailUnsubscribe.id.is_(None),
+                )
+            ).all()
```

Comment:
The Celery task doesn't have to do any exception handling for this DB query: if anything goes wrong an exception will be raised and Celery will catch the exception and retry the task after a delay